### PR TITLE
GitHub Actions for Elixir project with fixed caching

### PR DESCRIPTION
**Pull request by georgiybykov**
- Turn on caching for CI in GitHub Actions due to solved issue referenced in previous PR.
- Fix dialyzer warning.

### CODE REVIEW

File: .dialyzer-ignore.exs [hunk 1]
```diff
@@ -1,5 +1,3 @@
-# The list of the files to ignoge could be placed below.
+# The list of the warnings to ignore could be placed below.
 #   SEE: https://github.com/jeremyjh/dialyxir#elixir-term-format
-[
-  {"lib/kv_behaviour.ex", :unknown_type}
-]
+[]
```
Comment (am-kantox) [inline .dialyzer-ignore.exs:3]: 🎉

File: lib/kv_behaviour.ex [hunk 2]
```diff
@@ -1,10 +1,13 @@
 defmodule Kv.Behaviour do
   @moduledoc false
 
+  @typedoc "Key-value storage"
+  @type kv :: %{optional(any()) => any()}
```
Comment (am-kantox) [inline lib/kv_behaviour.ex:5]: ❤

**Review by am-kantox — APPROVED** [review]
:zap: :droplet: :zap: :gem: :zap: :blue_heart: :zap:

File: lib/kv_behaviour.ex (hunk 2 above)
Comment (georgiybykov) [reply lib/kv_behaviour.ex:5]: Eventually I found a way to fix the warning  :tada:

**Pull request merged**